kate/recovery/src/commitments.rs

```diff
@@ -98,44 +96,134 @@ pub fn verify_equality(
 	commitments: &[u8],
 	rows: &[Option<Vec<u8>>],
 	index: &index::AppDataIndex,
-	dimension: &matrix::Dimensions,
+	dimensions: &matrix::Dimensions,
 	app_id: u32,
-) -> Result<bool, Error> {
-	if commitments.len() / config::COMMITMENT_SIZE != rows.len() {
-		return Err(Error::InvalidData(DataError::RowAndCommitmentsMismatch));
-	}
-
+) -> Result<(Vec<u32>, Vec<u32>), Error> {
 	if commitments.len() % config::COMMITMENT_SIZE > 0 {
 		return Err(Error::InvalidData(DataError::BadCommitmentsData));
 	}
 
-	let app_rows = com::app_specific_rows(index, dimension, app_id);
+	if <u32>::try_from(commitments.len() / config::COMMITMENT_SIZE)? != dimensions.extended_rows() {
```

Comment:
Isn't that platform dependent and wouldn't `as` panic if 32-bit is not supported (should we even consider this case?). If we assume that smallest possible usize is u32 then 👍

Reply:
Yes, it is platform dependent. It could be `u32` (like `WASM`) or `u64` (like current micros). 
See https://doc.rust-lang.org/std/primitive.usize.html

Reply:
Changed to
`if (commitments.len() / config::COMMITMENT_SIZE) != dimensions.extended_rows().try_into()?`
to cast to bigger type and avoid panics. Wdyt?